src/language_features/texlab.rs

```diff
@@ -0,0 +1,95 @@
+use crate::context::Context;
+use crate::position::get_lsp_position;
+use crate::types::{EditorMeta, EditorParams};
+use crate::PositionParams;
+use lsp_types::request::Request;
+use lsp_types::TextDocumentIdentifier;
+use lsp_types::TextDocumentPositionParams;
+use lsp_types::Url;
+use serde::{Deserialize, Serialize};
+
+pub enum ForwardSearch {}
+
+impl Request for ForwardSearch {
+    type Params = TextDocumentPositionParams;
+    type Result = ForwardSearchResult;
+    const METHOD: &'static str = "textDocument/forwardSearch";
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ForwardSearchResult {
+    status: ForwardSearchStatus,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub enum ForwardSearchStatus {
+    Success = 0,
+    Error = 1,
+    Failure = 2,
+    Unconfigured = 3,
+}
+
+pub fn forward_search(meta: EditorMeta, params: EditorParams, ctx: &mut Context) {
+    let params = PositionParams::deserialize(params).unwrap();
+    let req_params = TextDocumentPositionParams {
+        text_document: TextDocumentIdentifier {
+            uri: Url::from_file_path(&meta.buffile).unwrap(),
+        },
+        position: get_lsp_position(&meta.buffile, &params.position, ctx).unwrap(),
+    };
+    ctx.call::<ForwardSearch, _>(meta, req_params, move |ctx, meta, response| {
+        forward_search_response(meta, response, ctx)
+    });
+}
+
+pub fn forward_search_response(meta: EditorMeta, result: ForwardSearchResult, ctx: &mut Context) {
+    let command = format!("echo {:?}", result);
+    ctx.exec(meta, command);
+}
+
+pub enum Build {}
+
+impl Request for Build {
+    type Params = BuildTextDocumentParams;
+    type Result = BuildResult;
+    const METHOD: &'static str = "textDocument/build";
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct BuildTextDocumentParams {
+    text_document: TextDocumentIdentifier,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct BuildResult {
+    status: BuildStatus,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub enum BuildStatus {
```

Comment:
This is plain wrong, this should derive `Serialize_repr`  and `Deserialize_repr` with `#[repr(i32)]`.